### ISSUE TITLE
pip install fails (Python 3)

### ISSUE THREAD

**Issue by ax3l** [dpgrote/Forthon#6]
Hi Dave,

I am trying to install Forthon on a Ubuntu 14.04 server (gfortran 4.8.2) and fail on

``` bash
$ pip [-v] install Forthon
Collecting Forthon
  Downloading Forthon-0.8.19.tar.gz (83kB)
...
```

with

```
running build_scripts
creating build/scripts-3.5
error: file '/tmp/pip-build-52frweiy/Forthon/Forthon3' does not exist
```

Environment:
- pip 8.1.1
- Anaconda 4.0
- Python 3.5.1

The verbose output is

```
Collecting Forthon
  1 location(s) to search for versions of Forthon:
  * https://pypi.python.org/simple/forthon/
  Getting page https://pypi.python.org/simple/forthon/
  Looking up "https://pypi.python.org/simple/forthon/" in the cache
  Current age based on date: 437
  Freshness lifetime from max-age: 600
  Freshness lifetime from request max-age: 600
  The response is "fresh", returning cached response
  600 > 437
  Analyzing links from page https://pypi.python.org/simple/forthon/
    Found link https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.11.tar.gz#md5=fa77ad60646f78e176293989f8be3c89 (from https://pypi.python.org/simple/forthon/), version: 0.8.11
    Found link https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.13.tar.gz#md5=b0d29aba88448270e7e2ea1bf30f2d4c (from https://pypi.python.org/simple/forthon/), version: 0.8.13
    Found link https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.14.tar.gz#md5=251c83781e84bd29dfcff2e7af7e74e6 (from https://pypi.python.org/simple/forthon/), version: 0.8.14
    Found link https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.15.tar.gz#md5=0973c39167911547c91d8395daecac1f (from https://pypi.python.org/simple/forthon/), version: 0.8.15
    Found link https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.16.tar.gz#md5=60cf6e312b9c6a5904145fc1dddb9689 (from https://pypi.python.org/simple/forthon/), version: 0.8.16
    Found link https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.17.tar.gz#md5=790acd517d22120fa92a5a42c18783f2 (from https://pypi.python.org/simple/forthon/), version: 0.8.17
    Found link https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.18.tar.gz#md5=7895705bfaf91882f7feed750abc32b5 (from https://pypi.python.org/simple/forthon/), version: 0.8.18
    Found link https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.19.tar.gz#md5=17f6a6cf058f0cf457ac02cdb1090e34 (from https://pypi.python.org/simple/forthon/), version: 0.8.19
  Using version 0.8.19 (newest of versions: 0.8.11, 0.8.13, 0.8.14, 0.8.15, 0.8.16, 0.8.17, 0.8.18, 0.8.19)
  Looking up "https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.19.tar.gz" in the cache
  Current age based on date: 437
  Freshness lifetime from max-age: 31557600
  The response is "fresh", returning cached response
  31557600 > 437
  Using cached Forthon-0.8.19.tar.gz
  Downloading from URL https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.19.tar.gz#md5=17f6a6cf058f0cf457ac02cdb1090e34 (from https://pypi.python.org/simple/forthon/)
  Running setup.py (path:/tmp/pip-build-r142h5n1/Forthon/setup.py) egg_info for package Forthon
    Running command python setup.py egg_info
    running egg_info
    creating pip-egg-info/Forthon.egg-info
    writing dependency_links to pip-egg-info/Forthon.egg-info/dependency_links.txt
    writing pip-egg-info/Forthon.egg-info/PKG-INFO
    writing top-level names to pip-egg-info/Forthon.egg-info/top_level.txt
    writing manifest file 'pip-egg-info/Forthon.egg-info/SOURCES.txt'
    warning: manifest_maker: standard file '-c' not found

    reading manifest file 'pip-egg-info/Forthon.egg-info/SOURCES.txt'
    writing manifest file 'pip-egg-info/Forthon.egg-info/SOURCES.txt'
  Source in /tmp/pip-build-r142h5n1/Forthon has version 0.8.19, which satisfies requirement Forthon from https://pypi.python.org/packages/source/F/Forthon/Forthon-0.8.19.tar.gz#md5=17f6a6cf058f0cf457ac02cdb1090e34
Building wheels for collected packages: Forthon
  Running setup.py bdist_wheel for Forthon ...   Destination directory: /tmp/tmppwwj9v58pip-wheel-
  Running command /home/huebl/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-r142h5n1/Forthon/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d /tmp/tmppwwj9v58pip-wheel- --python-tag cp35
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/Forthon
  copying source/cfinterface.py -> build/lib/Forthon
  copying source/Forthon_builder.py -> build/lib/Forthon
  copying source/wrappergen_derivedtypes.py -> build/lib/Forthon
  copying source/interfaceparser.py -> build/lib/Forthon
  copying source/fvars.py -> build/lib/Forthon
  copying source/Forthon_options.py -> build/lib/Forthon
  copying source/ForthonTimer.py -> build/lib/Forthon
  copying source/compilers.py -> build/lib/Forthon
  copying source/wrappergenerator.py -> build/lib/Forthon
  copying source/_Forthon.py -> build/lib/Forthon
  copying source/preprocess.py -> build/lib/Forthon
  copying source/__init__.py -> build/lib/Forthon
  copying source/version.py -> build/lib/Forthon
  copying source/License.txt -> build/lib/Forthon
  copying source/Forthon.h -> build/lib/Forthon
  copying source/Forthon.c -> build/lib/Forthon
  Skipping optional fixer: buffer
  Skipping optional fixer: idioms
  Skipping optional fixer: set_literal
  Skipping optional fixer: ws_comma
  running build_scripts
  creating build/scripts-3.5
  error: file '/tmp/pip-build-r142h5n1/Forthon/Forthon3' does not exist
error
  Failed building wheel for Forthon
  Running setup.py clean for Forthon
  Running command /home/huebl/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-r142h5n1/Forthon/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" clean --all
  running clean
  'build/lib' does not exist -- can't clean it
  'build/bdist.linux-x86_64' does not exist -- can't clean it
  'build/scripts-3.5' does not exist -- can't clean it
Failed to build Forthon
Installing collected packages: Forthon
  Running setup.py install for Forthon ...     Running command /home/huebl/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-r142h5n1/Forthon/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-e0ji5pjp-record/install-record.txt --single-version-externally-managed --compile
    running install
    running build
    running build_py
    creating build
    creating build/lib
    creating build/lib/Forthon
    copying source/cfinterface.py -> build/lib/Forthon
    copying source/Forthon_builder.py -> build/lib/Forthon
    copying source/wrappergen_derivedtypes.py -> build/lib/Forthon
    copying source/interfaceparser.py -> build/lib/Forthon
    copying source/fvars.py -> build/lib/Forthon
    copying source/Forthon_options.py -> build/lib/Forthon
    copying source/ForthonTimer.py -> build/lib/Forthon
    copying source/compilers.py -> build/lib/Forthon
    copying source/wrappergenerator.py -> build/lib/Forthon
    copying source/_Forthon.py -> build/lib/Forthon
    copying source/preprocess.py -> build/lib/Forthon
    copying source/__init__.py -> build/lib/Forthon
    copying source/version.py -> build/lib/Forthon
    copying source/License.txt -> build/lib/Forthon
    copying source/Forthon.h -> build/lib/Forthon
    copying source/Forthon.c -> build/lib/Forthon
    Skipping optional fixer: buffer
    Skipping optional fixer: idioms
    Skipping optional fixer: set_literal
    Skipping optional fixer: ws_comma
    running build_scripts
    creating build/scripts-3.5
    error: file '/tmp/pip-build-r142h5n1/Forthon/Forthon3' does not exist
error
Cleaning up...
  Removing source in /tmp/pip-build-r142h5n1/Forthon
Command "/home/huebl/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-r142h5n1/Forthon/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-e0ji5pjp-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-r142h5n1/Forthon/
Exception information:
Traceback (most recent call last):
  File "/home/huebl/anaconda3/lib/python3.5/site-packages/pip/basecommand.py", line 209, in main
    status = self.run(options, args)
  File "/home/huebl/anaconda3/lib/python3.5/site-packages/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "/home/huebl/anaconda3/lib/python3.5/site-packages/pip/req/req_set.py", line 732, in install
    **kwargs
  File "/home/huebl/anaconda3/lib/python3.5/site-packages/pip/req/req_install.py", line 884, in install
    spinner=spinner,
  File "/home/huebl/anaconda3/lib/python3.5/site-packages/pip/utils/__init__.py", line 718, in call_subprocess
    % (command_desc, proc.returncode, cwd))
pip.exceptions.InstallationError: Command "/home/huebl/anaconda3/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-r142h5n1/Forthon/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-e0ji5pjp-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-r142h5n1/Forthon/
```

Can you help?

Thanks a lot,
Axel

**Comment by ax3l** [comment]
As an update:

I gave the **python 2.7** version a shot for now
- gfortran 4.9.2
- pip 8.1.1
- Anaconda 4.0.5
- Python 2.7.11
- Forthon 0.8.19

and it worked :)

update: Python3.4 install from source via `setup.py` works, too.
seems really to be a pip problem (tried it again outside of conda with `pip install Forthon --user`)

**Comment by ax3l** [comment]
If you want to have a look at the failure yourself, I added a compile matrix under the following link:
  https://travis-ci.org/ax3l/Forthon/builds/122974947

**Comment by dpgrote** [comment]
Hi Axel - Were you able to install Forthon with this kind of setup before? If so, which version? Or, is this a new setup?

**Comment by ax3l** [comment]
Hi Dave, I remember installing Forthon with pip3 did not work last time I checked (2015) either and I used the source-install in that case too, so I guess it is actually broken for pip + python3 installs (also confirmed with the ubuntu-travis-ci builds I linked).

**Comment by ax3l** [comment]
fixed with 819bc7a3279ed321a31ed992d34db0a9e454c7fc, thank you! :sparkles:
